### PR TITLE
[export] Remove unused constants

### PR DESCRIPTION
**Pull request by angelayi**
An internal test case ran into a weird issue when exporting, where the model imported a file which creates tensor constants upon importing [(code ptr)](https://fburl.com/code/xwmhxm7n). This causes the tracer to create some tensor constants even though it's not used in the model code. This PR updates the lift_constant_tensors pass to remove constant nodes that are not being used instead of lifting them as tensor constants.

### CODE REVIEW

**Review by dolpm — APPROVED** [review]

File: torch/_export/passes/lift_constants_pass.py [hunk 1]
```diff
@@ -110,6 +110,55 @@ def _get_first_fqn(
...
+            torch.ops.aten.detach_.default,
+            torch.ops.aten.detach.default,
+        )
+    ):
+        return None
+
+    if len(detach_node.users) > 0:
+        return None
+    else:
+        return [detach_node, lift_fresh_node, node]
```
Comment (pianpwk) [inline torch/_export/passes/lift_constants_pass.py:159]: kinda nitpicking but this wouldn't match for something dumb like `y = torch.tensor(3)`? Would it be feasible to detect this in DCE with a custom impure function that matches detach nodes instead? Or is that besides the point.

Comment (angelayi) [reply torch/_export/passes/lift_constants_pass.py:159]: Yup that's true but somehow we specifically marked detach as being impure? https://www.internalfb.com/code/fbsource/[dac0d6c5b4bd]/fbcode/caffe2/torch/export/_trace.py?lines=1717-1720

Comment (pianpwk) [reply torch/_export/passes/lift_constants_pass.py:159]: can we add a todo? feels like an incomplete solution

**Review by pianpwk — APPROVED** [review]
